app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -12,35 +12,108 @@ import androidx.compose.material3.Scaffold
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
 import com.wire.android.R
 import com.wire.android.ui.common.bottomsheet.MenuBottomSheetItem
 import com.wire.android.ui.common.bottomsheet.MenuItemIcon
 import com.wire.android.ui.common.bottomsheet.MenuModalSheetLayout
 import com.wire.android.ui.common.snackbar.SwipeDismissSnackbarHost
+import com.wire.android.ui.common.WireDialog
+import com.wire.android.ui.common.WireDialogButtonProperties
+import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.ui.common.button.WireButtonState
 import com.wire.android.ui.home.conversations.mock.mockMessages
 import com.wire.android.ui.home.conversations.model.AttachmentBundle
 import com.wire.android.ui.home.conversations.model.Message
 import com.wire.android.ui.home.conversations.model.MessageSource
 import com.wire.android.ui.home.messagecomposer.MessageComposer
 import kotlinx.coroutines.launch
+import com.wire.android.util.dialogErrorStrings
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ConversationScreen(
     conversationViewModel: ConversationViewModel
 ) {
     val uiState = conversationViewModel.conversationViewState
-
+    val deleteMessageDialogState = conversationViewModel.deleteMessageDialogsState
     ConversationScreen(
         conversationViewState = uiState,
         onMessageChanged = { message -> conversationViewModel.onMessageChanged(message) },
         onSendButtonClicked = { conversationViewModel.sendMessage() },
         onSendAttachment = { attachmentBundle -> conversationViewModel.sendAttachmentMessage(attachmentBundle) },
-        onBackButtonClick = { conversationViewModel.navigateBack() }
+        onBackButtonClick = { conversationViewModel.navigateBack() },
+        onDeleteMessage = conversationViewModel::showDeleteMessageDialog
     )
+    if (deleteMessageDialogState is DeleteMessageState.State) {
```

Comment:
confusing if check, not really sure what would it mean to check if a state object is a state object ?

Reply:
Did some renaming, I will get rid of this when refactoring the dialog totally!

Reply:
I still find it confusing how the state of the dialogs is modelled, maybe it it just me but I think it would be less confusing if we did it some like this : 

it looks to me like all  the dialogs are kind of seperated @Composable,elements ViewModel  could hold the state for each of them, I do no think you need to do somehing like this : 
```

sealed class DeleteMessageState {
    data class State(
        val deleteMessageDialogState: DeleteMessageDialogState,
        val deleteMessageForYourselfDialogState: DeleteMessageDialogState
    ) : DeleteMessageState()

    data class Error(val coreFailure: CoreFailure) : DeleteMessageState()
}

```

you could hold those states as seperated variables, something like that for example : 

    val deleteMessageDialogState: DeleteMessageDialogState : DeleteMessageState()
   val deleteMessageForYourselfDialogState: DeleteMessageDialogStat   : DeleteMessageState()
   val error: DeleteMessageError : DeleteMessageError()

then you could pass those variables as a input to a composable  for example to : 

` DeleteMessageDialog()`
       
       and from there you handle the UI state for that individual dialog
       
       i think it would be abit more compose way and that way you do not need to duplicate code for eroor dialog and you also remove that additional if check as well as that nested when loop
       
      hope that makes sense and I am not telling you some bullshit :P as perhaps you have some reason to model the states as you did, it is just for me abit confusing